pkg/pool-linear/contracts/LinearPool.sol

```diff
@@ -512,21 +493,15 @@ abstract contract LinearPool is ILinearPool, IGeneralPool, IRateProvider, BasePo
         return (bptAmountIn, amountsOut);
     }
 
-    function _getMaxTokens() internal pure override returns (uint256) {
-        return _TOTAL_TOKENS;
-    }
-
     function _getMinimumBpt() internal pure override returns (uint256) {
         // Linear Pools don't lock any BPT, as the total supply will already be forever non-zero due to the preminting
         // mechanism, ensuring initialization only occurs once.
         return 0;
     }
```

Comment:
I know this was part of the original design, but would consider opening an issue to remove this. It feels like an unnecessary optimization that adds complexity.

Reply:
Created https://github.com/balancer-labs/balancer-v2-monorepo/issues/1958